modules/RFID/src/mysql.cc

```
@@ -0,0 +1,88 @@
+/**
+ * \file      mysql.cc
+ * \brief     Mysql library with error checking that uses the mysqlcppconn library
+ * \author    Tim IJntema
+ * \copyright Copyright (c) 2017, The R2D2 Team
+ * \license   See LICENSE
+ */
+
+#include "mysql.hh"
+
+MySql::MySql(){
+    driver = get_driver_instance();
+}
+
+MySql::~MySql(){
+    delete result;
+    delete statement;
+    delete connection;
+}
+
+template <typename T>
+bool MySql::errorCheck(T function){
+    try{
+        function();
+    } catch (sql::SQLException & error) {
+        std::cerr << "MySQL error code: " << error.getErrorCode()
+                  << ", SQLState: " << error.getSQLState() << '\n';
+        return false;
+    }
+    return true;
```

What's the point of discarding error information?
If the caller doesn't care if something succeeded or not, you might as well create an `ignoreErrors(T)` function.
Otherwise, allow the caller to handle errors by *not* reducing it to a boolean and rethrowing the exception instead, so the caller has more information.